### PR TITLE
Add ludwig model datatype

### PR DESCRIPTION
**Pull request by qiagu**
(Please replace this header with a description of your pull request. Please include *BOTH* what you did and why you made the changes. The "why" may simply be citing a relevant Galaxy issue.)
(If fixing a bug, please add any relevant error or traceback)
(For UI components, it is recommended to include screenshots or screencasts)

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/datatypes/binary.py [hunk 1]
```diff
@@ -1611,6 +1612,37 @@ def display_data(self, trans, dataset, preview=False, filename=None, to_ext=None
...
+        rval.append('<div>This model dataset is composed of the following items:<p/><ul>')
+        for composite_name, composite_file in self.get_composite_files(dataset=dataset).items():
+            fn = composite_name
+            opt_text = ''
+            if composite_file.optional:
+                opt_text = ' (optional)'
+            if composite_file.get('description'):
+                rval.append(f"<li><a href=\"{fn}\" type=\"text/plain\">{fn} ({composite_file.get('description')})</a>{opt_text}</li>")
+            else:
+                rval.append(f'<li><a href="{fn}" type="text/plain">{fn}</a>{opt_text}</li>')
```
Comment (jdavcs) [inline lib/galaxy/datatypes/binary.py]: Maybe simplify to something like that? (or something along these lines). That way we have only one parametrized version of the html we are building.

```suggestion
            description = composite_file.get('description')
            link_text = f'{composite_name} ({description})' if description else composite_name
            opt_text = ' (optional)' if composite_file.optional else ''
            rval.append(f'<li><a href="{composite_name}" type="text/plain">{link_text}</a>{opt_text}</li>')
```

Comment (qiagu) [reply lib/galaxy/datatypes/binary.py]: Super!

**Review by jdavcs — APPROVED** [review]
Thanks for applying the suggested change! 
All looks good!

**Comment by qiagu** [comment]
Thanks, @mvdbeek, @ic4f

**Pull request merged**